### PR TITLE
fix: Frontend related issues and warnings

### PR DESCRIPTION
**Pull request by ayushjain17**
## Problem

1. `dlmalloc` failing in FE after upgrading to rust 1.78.0
2. unhandled warnings use of signals in non-reactive sections
3. content mismatch issue while using key grouping
4. range for ramp is allowed till 100

## Solution

1. Revert back to rust 1.76.0
2. convert `get` to `get_untracked` and other similar fixes (might not have solved all)
3. move the check of local storage value to a `create_effect` to avoid initial value mismatch between client and SSR
4. limit the max value of ramp based on the allowed max value

### CODE REVIEW

File: crates/frontend/src/pages/experiment_list.rs [hunk 1]
```diff
@@ -177,7 +177,7 @@ pub fn experiment_list() -> impl IntoView {
                                                     cell_class="min-w-48 font-mono".to_string()
                                                     rows=data
                                                     key_column="id".to_string()
-                                                    columns=table_columns.get()
+                                                    columns=table_columns.get_untracked()
```
Comment (ShubhranshuSanjeev) [inline crates/frontend/src/pages/experiment_list.rs:180]: @ayushjain17 , thinking we can get rid of create_memo for table_columns?

File: crates/frontend/src/pages/home.rs [hunk 2]
```diff
@@ -303,8 +303,10 @@ pub fn home() -> impl IntoView {
         }
         // resolve the context and get the config that would apply
         spawn_local(async move {
-            let context = gen_query_context(context_rs.get());
-            let mut config = match resolve_config(tenant_rs.get(), context).await.unwrap()
+            let context = gen_query_context(context_rs.get_untracked());
```
Comment (ShubhranshuSanjeev) [inline crates/frontend/src/pages/home.rs:306]: Why replace with `get_untracked` here? If `get` is not reactive here then move the `get` call to outside of async block. Let's keep the `get`.

**Review by ShubhranshuSanjeev — CHANGES_REQUESTED** [review]

File: crates/frontend/src/components/default_config_form.rs [hunk 3]
```diff
@@ -141,7 +141,7 @@ where
                     type="text"
                     placeholder="Key"
                     class="input input-bordered w-full max-w-md"
-                    value=config_key.get()
+                    value=config_key.get_untracked()
```
Comment (Datron) [inline crates/frontend/src/components/default_config_form.rs:144]: Are we sure get_untracked is right here? Do we not want reactivity?

File: rust-toolchain.toml [hunk 4]
```diff
@@ -1,4 +1,4 @@
 [toolchain]
-channel = "1.78.0"
+channel = "1.76.0"
```
Comment (Datron) [inline rust-toolchain.toml:2]: Why is `dlmalloc` failing? We can debug that, rather than downgrading the rust version

**Review by Datron — CHANGES_REQUESTED** [review]

File: crates/frontend/src/components/default_config_form.rs (hunk 3 above)
Comment (ayushjain17) [reply crates/frontend/src/components/default_config_form.rs:144]: it was anyways not reactive earlier at this point, this just makes it explicit and avoids warnings

File: rust-toolchain.toml (hunk 4 above)
Comment (ayushjain17) [reply rust-toolchain.toml:2]: to fix this you need to upgrade wasm-bindgen to `2.92.0` or higher, and once you do that web-sys starts failing to compile and then we need to fix that then the chain continues

**Review by Datron — APPROVED** [review]

**Review by ShubhranshuSanjeev — APPROVED** [review]

**Pull request merged**